### PR TITLE
dwl: 0.3.1 -> 0.4

### PR DESCRIPTION
**Pull request by anas-z0**
change log: https://github.com/djpohly/dwl/issues/307

### CODE REVIEW

File: pkgs/applications/window-managers/dwl/default.nix [hunk 1]
```diff
@@ -8,9 +8,9 @@
 , pkg-config
 , wayland
 , wayland-protocols
-, wlroots
+, wlroots_0_16
 , writeText
-, enable-xwayland ? true, xwayland, libX11
+, enable-xwayland ? true, xorg, libX11
```
Comment (ckiee) [inline pkgs/applications/window-managers/dwl/default.nix:13]:
```suggestion
, enableXWayland ? true, xorg, libX11
```

enable-… is not nixpkgs style, and wlroots already spells it `enableXWayland` too.

File: pkgs/applications/window-managers/dwl/default.nix [hunk 2]
```diff
@@ -39,18 +39,18 @@ stdenv.mkDerivation rec {
...
   ] ++ lib.optionals enable-xwayland [
     libX11
-    xwayland
+    xorg.xcbutilwm
   ];
 
   # Allow users to set their own list of patches
   patches = totalPatches;
 
-  # Last line of config.mk enables XWayland
```
Comment (ckiee) [inline pkgs/applications/window-managers/dwl/default.nix:51]: This won't be fun if the file gets bigger and now `13-14` is in the middle. Can we:
```
cat <<EOF >config.mk
XWAYLAND = -DXWAYLAND
XLIBS = xcb xcb-icccm
EOF
```
Or something?

**Review by ckiee — CHANGES_REQUESTED** [review]

**Comment by AndersonTorres** [comment]
Squash the commits into one.

**Review by AndersonTorres — DISMISSED** [review]
Remember squash all commits into one.

**Comment by ckiee** [comment]
I think it's kind of a shame since this is their first PR and that could be discouraging.

**Comment by AndersonTorres** [comment]
This is not an easy case of "update source and hash".

The changes I did are a bit profound. Indeed I believe bigger and better changes are needed.

But if the greater objective is to be warm and cordial to beginners that prefer to use github web interface instead of the cold startling phosphor tube, OK, I will merge this.

... After an explanation of why the hell xwayland was removed from inputs.

**Pull request closed without merging**